pkg/providers/provider.go

```diff
@@ -46,6 +46,15 @@ type Provider interface {
 	PreCoreComponentsUpgrade(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error
 }
 
+// TemporaryProvider is a Provider but expanded to take on different interfaces for a few methods. This is a temporary change
+// to facilitate refactoring efforts in smaller changes.
+type TemporaryProvider interface {
```

Comment:
do we need a new interface? i think if we return the types, we don't
also, we can still return the old interface even if we don't return the type

what is this interface for? no one is consuming it

Reply:
This was added to add the new methods without changing the original interface first. They would not be accessible in the tests through `provider.Provider`. Perhaps, we can export the docker `provider` struct that implements `provider.Provider` and reference that in the tests instead?